loadbalancer/algorithm.go

```diff
@@ -168,32 +72,30 @@ func (r *roundRobin) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 		return ctx.Route.LBEndpoints[0]
 	}
 
+	ctx.Route.Lock()
+	defer ctx.Route.Unlock()
```

Comment:
Here I am a bit worried about the 2-phase lock at high traffic routes.
What do you think @AlexanderYastrebov ?

Reply:
Here we lock only to check whether ctx.Route.GoodEndpoints is empty.
Maybe we should juts have 
```go
func (r *Route) GetHealthyEndpoints() []LBEndpoint
```
that would take care of proper locking internally.
It may also guarantee to never return an empty list.